felix/daemon/daemon.go

```diff
@@ -642,16 +643,32 @@ configRetry:
 		delay := configParams.EndpointReportingDelaySecs
 		log.WithField("delay", delay).Info(
 			"Endpoint status reporting enabled, starting status reporter")
-		dpConnector.statusReporter = statusrep.NewEndpointStatusReporter(
+
+		fromDataplaneC := make(chan interface{})
+		dpConnector.StatusUpdatesFromDataplaneConsumers = append(dpConnector.StatusUpdatesFromDataplaneConsumers, fromDataplaneC)
+		statusReporter := statusrep.NewEndpointStatusReporter(
 			configParams.FelixHostname,
 			configParams.OpenstackRegion,
-			dpConnector.StatusUpdatesFromDataplane,
+			fromDataplaneC,
 			dpConnector.InSync,
 			dpConnector.datastore,
 			delay,
 			delay*180,
 		)
-		dpConnector.statusReporter.Start()
+		statusReporter.Start()
+	}
+
+	if configParams.EndpointStatusPathPrefix != nil && *configParams.EndpointStatusPathPrefix != "" {
+		fromDataplaneC := make(chan interface{})
+		dpConnector.StatusUpdatesFromDataplaneConsumers = append(dpConnector.StatusUpdatesFromDataplaneConsumers, fromDataplaneC)
+		statusFileReporter := statusrep.NewEndpointStatusFileReporter(fromDataplaneC, *configParams.EndpointStatusPathPrefix)
+
+		// TODO: investigate if it's a requirement that
```

Comment:
These components should never stop.  Felix will restart if the config changed.  We only use contexts here for easier testing, I think.